### PR TITLE
[BUG] Fix Discover table panel size auto adjust in Dashboard

### PR DESCRIPTION
**Pull request by ananzh**
### Description
* Add React.memo on parent comp
* Restore embeddable panel style

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5415

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/79961084/5a8ad865-d179-41b7-a49b-e0a269e7a1ed

### Check List

- [ ] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ananzh** [comment]
For the failed cypress tests

* ciGroup6

Screenshot 2023-11-08 at 12 07 59

This error is because after set 2s as auto refresh interval, the superDatePicker menu panel is closed, which causes the button unable to find. This is not caused by the fix.

Screenshot 2023-11-08 at 12 09 03

File: CHANGELOG.md [hunk 1]
```diff
@@ -76,6 +76,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [BUG] Add platform "darwin-arm64" to unit test ([#5290](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5290))
 - [BUG][Dev Tool] Add dev tool documentation link to dev tool's help menu [#5166](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5166)
 - Fix navigation issue across dashboards ([#5435](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5435))
+- [BUG] Fix Discover table panel size auto adjust in Dashboard ([#5441](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5441))
```
Comment (AMoo-Miki) [inline CHANGELOG.md:79]:
```suggestion
- [Discover] Fix table panel auto-sizing ([#5441](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5441))
```

File: src/plugins/discover/public/embeddable/search_embeddable_component.tsx [hunk 2]
```diff
@@ -51,8 +52,10 @@ export function SearchEmbeddableComponent({ searchProps }: SearchEmbeddableProps
         data-test-subj="embeddedSavedSearchDocTable"
       >
         {discoverEmbeddableProps.totalHitCount !== 0 ? (
-          <EuiFlexItem>
-            <DataGridTableMemoized {...discoverEmbeddableProps} />
+          <EuiFlexItem style={{ minHeight: 0 }}>
+            <div className="osdDocTable__container">
```
Comment (AMoo-Miki) [inline src/plugins/discover/public/embeddable/search_embeddable_component.tsx:56]: Why not just give the class name to the flex item?

Comment (ananzh) [reply src/plugins/discover/public/embeddable/search_embeddable_component.tsx]: Good call. I thought EuiFlexItem is using `displayName`. Apparently, I remembered this wrong. I fixed the comment.

**Review by AMoo-Miki — APPROVED** [review]

**Review by manasvinibs — APPROVED** [review]

**Pull request merged**